pkg/proxy/util/utils.go

```diff
@@ -221,29 +221,22 @@ func GetNodeAddresses(cidrs []string, nw NetworkInterfacer) (sets.String, error)
 		}
 
 		_, ipNet, _ := net.ParseCIDR(cidr)
-		for _, itf := range itfs {
-			addrs, err := nw.Addrs(&itf)
-			if err != nil {
-				return nil, fmt.Errorf("error getting address from interface %s, error: %v", itf.Name, err)
+		for _, addr := range addrs {
+			if addr == nil {
+				continue
 			}
 
-			for _, addr := range addrs {
-				if addr == nil {
-					continue
-				}
+			ip, ok := addr.(*net.IPNet)
+			if !ok {
+				return nil, fmt.Errorf("error corvert addr to net.IPNet for address %s, error: %v", addr.String(), err)
```

Comment:
I couldn't investigate more, the unit tests fail here casting the IPs, however, the e2e test doesn't show any error